python/test/unit/fem/test_petsc_nonlinear_assembler.py

```diff
@@ -391,76 +271,48 @@ def initial_guess_p(x):
 
         def blocked_solve():
             """Blocked version"""
-            Jmat = create_matrix_block(J)
-            Fvec = create_vector_block(F)
-            snes = PETSc.SNES().create(MPI.COMM_WORLD)
-            snes.setTolerances(rtol=1.0e-15, max_it=10)
-            problem = NonlinearPDE_SNESProblem(F, J, [u, p], bcs)
-            snes.setFunction(problem.F_block, Fvec)
-            snes.setJacobian(problem.J_block, J=Jmat, P=None)
-
             u.interpolate(initial_guess_u)
             p.interpolate(initial_guess_p)
 
-            x = create_vector_block(F)
-            scatter_local_vectors(
-                x,
-                [u.x.petsc_vec.array_r, p.x.petsc_vec.array_r],
-                [
-                    (u.function_space.dofmap.index_map, u.function_space.dofmap.index_map_bs),
-                    (p.function_space.dofmap.index_map, p.function_space.dofmap.index_map_bs),
-                ],
+            snes_options = {"snes_rtol": 1.0e-15, "snes_max_it": 10, "snes_monitor": None}
+            snes, x = dolfinx.nls.petsc.create_snes_solver(
```

Comment:
In addition, could we see an example of use/test where the very lowest level components are used outside of `create_snes_solver`? I.e. the user sets up SNES manually. I'd like to see how much boiler plate that involves.

Reply:
I can do this.